processing/src/main/java/org/apache/druid/query/groupby/orderby/DefaultLimitSpec.java

```diff
@@ -220,6 +221,15 @@ public boolean isLimited()
       sortingNeeded = !query.getGranularity().equals(Granularities.ALL) && query.getContextSortByDimsFirst();
     }
 
+    if (!sortingNeeded) {
+      Map<String, Object> timestampFieldContext = GroupByQueryHelper.findTimestampResultField(query);
```

Comment:
Why not finding the index from the query context?

Reply:
The whole timestampResult optimization doesn't affect the original query context now, the modification to `DruidQuery `has been withdraw, currently only `GroupByStrategyV2` is actually modified.